extension/src/plots/model/quickPick.ts

```diff
@@ -99,9 +100,54 @@ export const pickCustomPlotType = (): Thenable<CustomPlotType | undefined> => {
 const getTypeColumnLikes = (columns: Column[], columnType: ColumnType) =>
   columns
     .filter(({ type }) => type === columnType)
-    .map(({ label, path }) => ({ label, path }))
+    .map(({ path, label }) => ({ label, path }))
 
-export const pickMetricAndParam = async (columns: Column[]) => {
+const getCustomPlotIds = (
+  customPlotVals: CustomPlotsOrderValue[],
+  customPlotType: CustomPlotType
+) => {
+  const plotIds = new Set()
+
+  for (const { type, metric, param } of customPlotVals) {
+    if (type === customPlotType) {
+      plotIds.add(getCustomPlotId(metric, param))
+    }
+  }
+
+  return plotIds
+}
+
+const filterColumnsBasedOffCustomPlotValues = (
+  columns: Column[],
+  customPlotVals: CustomPlotsOrderValue[],
+  customPlotType: CustomPlotType,
+  knownPlotMetric: string | undefined,
+  knownPlotParam?: string
+) => {
+  const typeToSortColumnsBy = isCheckpointValue(customPlotType)
+    ? ColumnType.METRICS
+    : ColumnType.PARAMS
+  const typeColumnLikes = getTypeColumnLikes(columns, typeToSortColumnsBy)
+  const plotIds = getCustomPlotIds(customPlotVals, customPlotType)
+
+  return plotIds.size > 0
+    ? typeColumnLikes.filter(({ path }) => {
+        const metric =
+          knownPlotMetric ||
+          removeColumnTypeFromPath(path, ColumnType.METRICS, FILE_SEPARATOR)
+        const param =
+          knownPlotParam ||
+          removeColumnTypeFromPath(path, ColumnType.PARAMS, FILE_SEPARATOR)
+
+        return !plotIds.has(getCustomPlotId(metric, param))
+      })
+    : typeColumnLikes
+}
+
+export const pickMetricAndParam = async (
+  columns: Column[],
+  customPlotOrderVals: CustomPlotsOrderValue[]
+) => {
```

Comment:
[I] In order to provide an accurate list of options I think you need to:

1. Generate the full params x metrics list but do not include entries that already exist.
2. Present the unique list of metrics from the list created by 1 in `pickFromColumnLikes` to get the metric. If there are no metrics left then you can show the initial warning.
3. Use the metric to filter the list created by 1 and pass the resulting params into `pickFromColumnLikes` to get the param.

This would mean you could never hit `Toast.showError('There are no params to select from.')`.

The initial list can be generated by something like
```
    const acc: { metric: string; param: string }[] = []
    for (const metric of metrics) {
      for (const param of params) {
        if (
          !existingOrder.includes({
            metric,
            param,
            type: CustomPlotType.METRIC_VS_PARAM
          })
        ) {
          acc.push({ metric, param })
        }
      }
    }

```